src-tauri/src/app.rs

```diff
@@ -177,20 +190,20 @@ impl Service {
             return Err(anyhow::anyhow!("channel tx closed"));
         };
 
-        self.context.lock().await.channel_set = true;
+        self.context.write().await.channel_set = true;
 
         Ok(channel)
     }
 }
 
 pub struct Rpc {
-    context: Arc<Mutex<Context>>,
+    pub(crate) context: Arc<RwLock<Context>>,
```

Comment:
I needed to make this accessible from the outside, otherwise I run into trouble moving the `Rpc` instance in the protocol handler. I'm sure there's a better way to do this, but it would need a refactor of the whole `Rpc` struct and I thought that's too much.